data_browser/src/ExperimentPage.js

```diff
@@ -119,62 +125,305 @@ function renderExperimentHeader(args) {
+    viewSingleUrl(trainerConfig.initialize_from) :
+    `random(${modelConfig.initializer_range})`;
+
+  const numSteps = trainerConfig.num_train_steps;
+  const batchSize = trainerConfig.train_batch_size;
+  const seqLen = modelConfig.seq_len;
+  const numTokens = computeNumTokens({numSteps, batchSize, seqLen});
+  const trainerSummary = <span>
+    {Math.round(numSteps)} (steps) * {renderStagedValue(batchSize)} (batch_size) * {modelConfig.seq_len} (seq_len) = {renderSciNotation(numTokens)} (tokens)
+  </span>;
+
+  // Hardware
+  const resources = step.config.resources;
+  const resourcesSummary = `${resources.slice_count || 1} * ${resources.tpu_type}`;
+
+  const wandbLink = <a href={getWandbUrl({step})} title="Go to the WandB page for this training run" target="_blank">[WandB {wandbIcon}]</a>;
+  const description = (
+    <table className="train-table">
+      <tbody>
+        <tr><td>Dataset:</td><td>{datasetSummary}</td></tr>
+        <tr><td>Architecture:</td><td>{architectureSummary}</td></tr>
+        <tr><td>Optimizer:</td><td>{optimizerSummary}</td></tr>
+        <tr><td>Initialization:</td><td>{initializationSummary}</td></tr>
+        <tr><td>Training:</td><td>{trainerSummary}</td></tr>
+        <tr><td>Hardware:</td><td>{resourcesSummary}</td></tr>
+        <tr><td>{wandbLink}</td></tr>
+      </tbody>
+    </table>
+  );
+  return {name: "train", description: description};
 }
 
-function renderExperimentStatus(events) {
-  if (!events) {
-    return "[Loading status]";
+function renderDatasetSummary({dataConfig, steps}) {
+  // sources: {source: {train_urls: [url1, url2, ...], ...}}
+  // weights:
+  // - {source: weight}
+  // - [[step, {source:weight}], [step, {source:weight}], ...]
+  const sources = dataConfig.configs;
+  const allWeights = Array.isArray(dataConfig.train_weights) ? dataConfig.train_weights : [[0, dataConfig.train_weights]];
+  // To normalize the weights
+  const sumWeights = allWeights.map(([step, weights]) => Object.values(weights).reduce((a, b) => a + b, 0));
+
+  const datasetRows = [];
+
+  // Header
+  if (allWeights.length > 1) {
+    const header = [<td key="source">Source \ Step</td>].concat(
+      allWeights.map(([step, weights], stage) => <td key={step}>{step}</td>)
+    );
+    datasetRows.push(<tr key="header">{header}</tr>);
   }
 
+  const nonzeroSources = Object.entries(sources).filter(([source, location]) => {
+    // This happens for validation sets (listed, but not trained on)
+    if (allWeights.every(([step, weights]) => weights[source] === 0)) {
+      return false;
+    }
+    return true;
+  });
+
+  if (nonzeroSources.length === 1) {
+    return renderPath({path: nonzeroSources[0][1].train_urls[0], steps});
+  }
+
+  nonzeroSources.forEach(([source, location]) => {
+    const row = [<td key="source">{renderPath({path: location.train_urls[0], steps})}</td>].concat(
+      allWeights.map(([step, weights], stage) => <td key={step}>{round(weights[source] / sumWeights[stage], 2)}</td>)
```

Comment:
@dlwh I took the liberty of normalizing - is the code invariant to scaling?

Reply:
yes